services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -161,4 +166,42 @@ class FlowValidationNegativeSpec extends BaseSpecification {
         def switches = topology.getActiveSwitches()
         return [switches.first(), switches.first()]
     }
+
+    @Unroll
+    def "Unable to #action flow on a non-existent flow"() {
+        when: "Try to #action rules on a non-existent flow"
+        northbound."$method"(NON_EXISTENT_FLOW_ID)
+
+        then: "An error is received (404 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Could not reroute flow: Flow non existent flow id non existent flow id not found"
+
+        where:
+        action        | method
+        "synchronize" | "synchronizeFlow"
+        "reroute"     | "rerouteFlow"
+
+    }
+
+    def "Unable to get flow on a non-existent flow"() {
+        when: "Try to synchronize rules on a non-existent flow"
```

Comment:
Please pay attention to step descriptions, they are important. Currently it has nothing in common with what is actually happening in the test. Same below